#!/usr/bin/env python
PACKAGE = "db_chassis"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_A",    double_t,    0, "Left motor P constant", 0.1,  0.0, 10.0)
gen.add("ki_A",    double_t,    0, "Left motor I constant", 0.0,  0.0, 1.0)
gen.add("kd_A",    double_t,    0, "Left motor D constant", 0.0,  0.0, 1.0)
gen.add("kp_B",    double_t,    0, "Right motor P constant", 0.1,  0.0, 10.0)
gen.add("ki_B",    double_t,    0, "Right motor I constant", 0.0,  0.0, 1.0)
gen.add("kd_B",    double_t,    0, "Right motor D constant", 0.0,  0.0, 1.0)
gen.add("speed_kA",    double_t,    0, "Left motor speed smoothing constant", 0.9,  0.0, 1.5)
gen.add("speed_kB",    double_t,    0, "Right motor speed smoothing constant", 0.9,  0.0, 1.5)

exit(gen.generate(PACKAGE, "db_chassis", "DodobotChassis"))
